corehq/apps/events/models.py

```diff
@@ -129,16 +199,65 @@ def _persist_event(self):
             'owner_id': self.manager.user_id,
             'properties': case_json,
         }
-        case_ = create_case_with_case_type(EVENT_CASE_TYPE, case_args)
+        case_ = create_case_with_case_type(self.EVENT_CASE_TYPE, case_args)
         self.case = case_
 
+    @classmethod
+    def _get_obj_from_data(cls, data) -> Event:
+        event_obj = cls()
 
-def get_domain_events(domain):
-    events = []
+        event_obj.domain = data['domain']
+        event_obj.name = data['name']
+        event_obj.start_date = data['start_date']
+        event_obj.end_date = data['end_date']
+        event_obj.attendance_target = data['attendance_target']
+        event_obj.sameday_reg = data['sameday_reg']
+        event_obj.track_each_day = data['track_each_day']
+        event_obj.manager = data['manager']
+        event_obj.is_open = data.get('is_open', Event.is_open)
+        event_obj.attendee_list_status = data.get('attendee_list_status', Event.attendee_list_status)
```

Comment:
Is it possible to use `cls(**data)` and modify the ones that are special to be more DRY?